examples/mnist_net2net.py

```diff
@@ -0,0 +1,381 @@
+  + Student model has an extra Conv2D layer and an extra FC layer.
+  + Comparison of "random-init" vs "net2deeper" weight initialization.
+  + With 'net2deeper', student model should immediately perform as well
+    as teacher model.
+- Hyper-parameters:
+  + Simple SGD is used intentionally to show the continuous learning
+    of student model, given that MNIST is a simple dataset.
+  + Learning rate adjustment: it's suggested to reduce learning rate
+    to 1/10 for student model, but not implemented here.
+  + Addition of noise in "net2wider" is used to break weight symmetry
+    and thus enable full capacity of student models. It is optional
+    when a Dropout layer is used.
+
+Results
+- Tested with "Theano" backend and "th" image_dim_ordering.
+- Running on GPU GeForce GTX 980M
+- Performance Comparisons - loss values during first 3 epochs:
+(1) teacher_model:             0.443    0.141    0.092
+(2) wider_random_pad:          0.098    0.058    0.047
+(3) wider_net2wider:           0.069    0.057    0.049
+(4) deeper_random_init:        0.170    0.068    0.049
+(5) deeper_net2deeper:         0.074    0.052    0.042
+'''
+
+from __future__ import print_function
+
+from keras.models import Sequential
+from keras.layers import Conv2D, MaxPooling2D, Dense, Flatten
+from keras.utils import np_utils
+from keras.datasets import mnist
+
+import numpy as np
+
+np.random.seed(1337)
+input_shape = (1, 28, 28)  # image shape
+nb_class = 10  # number of class
+
+# load and pre-process data
+
+(train_x, train_y), (validation_x, validation_y) = mnist.load_data()
+
+
+def preprocess_input(x):
+    return x.reshape((-1, ) + input_shape) / 255.
+
+
+def preprocess_output(y):
+    return np_utils.to_categorical(y)
+
+train_x, validation_x = map(preprocess_input, [train_x, validation_x])
+train_y, validation_y = map(preprocess_output, [train_y, validation_y])
+print("Loading MNIST data...")
+print("train_x shape:", train_x.shape, "train_y shape:", train_y.shape)
+print("validation_x shape:", validation_x.shape,
+      "validation_y shape", validation_y.shape)
+
+
+# knowledge transfer algorithms
+
+def wider2net_conv2d(teacher_w1, teacher_b1, teacher_w2, new_width, init):
+    """Get initial weights for a wider conv2d layer with a bigger nb_filter,
+    by 'random-padding' or 'net2wider'.
+
+    # Auguments
+        teacher_w1: `weight` of conv2d layer to become wider,
+          of shape (nb_filter1, nb_channel1, kh1, kw1)
+        teacher_b1: `bias` of conv2d layer to become wider,
+          of shape (nb_filter1, )
+        teacher_w2: `weight` of next connected conv2d layer,
+          of shape (nb_filter2, nb_channel2, kh2, kw2)
+        new_width: new `nb_filter` for the wider conv2d layer
+        init: initialization algorithm for new weights,
+          either 'random-pad' or 'net2wider'
+    """
+    assert teacher_w1.shape[0] == teacher_w2.shape[1], (
+        "successive layers from teacher model should have compatible shapes")
```

Comment:
You're using both `"` and `'` as the quote character, normalize to `'` everywhere.